source/tyr/tyr/binarisation.py

```diff
@@ -282,19 +299,22 @@ def osm2ed(self, instance_config, osm_filename, job_id, dataset_uid):
             cities_db = current_app.config.get('CITIES_DATABASE_URI')
             if not cities_db:
                 raise ValueError(
-                    'impossible to use osm2ed with cities db since no cities database configuration has been set'
+                    "impossible to use osm2ed with cities db since no cities database configuration has been set"
                 )
             args.extend(["--cities-connection-string", cities_db])
-        with collect_metric('osm2ed', job, dataset_uid):
-            res = launch_exec('osm2ed', args, logger)
+        with collect_metric("osm2ed", job, dataset_uid):
+            res = launch_exec("osm2ed", args, logger)
         if res != 0:
             # @TODO: exception
-            raise ValueError('osm2ed failed')
+            raise ValueError("osm2ed failed")
+        dataset.state = "done"
```

Comment:
Should this rather be the `ed2nav` job ?

Reply:
why?

Reply:
I thought your dataset would only be "done" when the `.nav` is generated at the end of all tasks. 
I'm maybe confused, but in this case, what's the difference between the `job.state` and `dataset.state` here ?

Reply:
When giving several types of data for binarization to Tyr, a task will be run for each type of data. Only one job is created for one binarization, with several datasets corresponding to each type of data (see description).
The `dataset.state` represents the state of the task `*2ed` associated to the dataset.
The `job.state` represents the state of each task `*2ed` + `ed2nav`

Now, if one task fails, we can know which one did.